src/libxtp/dftengine/dftengine.cc

```diff
@@ -380,6 +383,31 @@ namespace votca {
                 // DFT AOOverlap matrix
                 _dftAOoverlap.Fill(_dftbasis);
                 CTP_LOG(ctp::logDEBUG, *_pLog) << ctp::TimeStamp() << " Filled DFT Overlap matrix of dimension: " << _dftAOoverlap.Dimension() << flush;
+
+                if (_with_periodic_embedding) {
+                    // needs to be replaced by proper kvector setup function
+                    std::vector< vec > kvectors;
+                    kvectors.push_back(vec(1.0, 2.0, 3.0));
+                    kvectors.push_back(vec(-1.0, -2.0, -3.0));
+
+                    // store sum_g (V(G)*<a|exp(ikG)|b>) in _temp_PW_matrix, 
+                    // later store final result back into _AOPlanewave Matrix
+                    ub::matrix<std::complex<double>> _temp_PW_matrix = ub::zero_matrix<std::complex<double>>(_dftbasis.AOBasisSize());
+                    std::vector<vec>::iterator kit;
+
+                    for (kit = kvectors.begin(); kit != kvectors.end(); ++kit) {
+                        const vec k = *kit;
```

Comment:
why make a copy here instead of putting k into the k vectors, even easier is using the new c++11 range iteration feature 

``for (const auto& k:kvectors)``

Reply:
Can do, but again this is more a placeholder at the moment.